pkgs/development/compilers/crystal/default.nix

```diff
@@ -98,21 +100,37 @@ let
         inherit sha256;
       };
 
-      patches = lib.optionals (lib.versionOlder version "1.2.0") [
+      patches = [ ]
+        ++ lib.optionals (lib.versionOlder version "1.2.0") [
         # add support for DWARF5 debuginfo, fixes builds on recent compilers
         # the PR is 8 commits from 2019, so just fetch the whole thing
         # and hope it doesn't change
         (fetchpatch {
           url = "https://github.com/crystal-lang/crystal/pull/11399.patch";
           sha256 = "sha256-CjNpkQQ2UREADmlyLUt7zbhjXf0rTjFhNbFYLwJKkc8=";
         })
+      ] ++ lib.optionals (lib.versionAtLeast version "1.3.0" && lib.versionOlder version "1.6.0") [
+        # needed for #12601 to get applied successfully
+        (fetchpatch {
+          url = "https://github.com/crystal-lang/crystal/commit/bf4009dacbf67a63a1cbaffddbdc99549bb70a03.patch";
```

Comment:
That patch adds FFI support on Windows. What's the relevance of that?

Reply:
Couldn't apply https://github.com/crystal-lang/crystal/pull/12601 without it on < 1.6.0 (due to the `ffi_spec.cr` changes) - if there's a better alternative I'll gladly change it to that!

Reply:
Ah, okay that explains it then.

Not sure if all this backporting of fixes is really worth it, though. FFI bindings are only used for the interpreter, which is still only an experimental feature. That's why these fixes are not backported in upstream.

I'd recommend to just disable the interpreter (and FFI specs) in releases prior to 1.6.2. Even at this point the interpreter is known to have bugs.

Reply:
Interpreter is not enabled. If only the interpreter depends on ffi, then I guess I can pass `-Dwithout_ffi` instead of deleting the `ffi_spec`, right?

Reply:
Yes `-Dwithout_ffi` skips the FFI specs.

Reply:
I'm also wondering why specs are built in release mode. That's not really necessary and less efficient.

Reply:
Both should be done now:
- release mode only for the compiler: https://github.com/NixOS/nixpkgs/pull/195606/commits/5564346d68f6696e18dab41cf3546325e7b4647a
- disable ffi for < 1.6.1: https://github.com/NixOS/nixpkgs/pull/195606/commits/b1be734009b3b9aa217d8561846fce9c9beafb37